src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+                }
+                for (IBond tmpConjBond: tmpConjAtomContainer.bonds()) {
+                    int tmpBondInteger = tmpConjBond.getProperty(tmpAlkylSFBondIndexProperty);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISCONJUGATED,true);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISPLACED,false);
+                }
+            }
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING,
+                    anException + " Molecule ID: " + aMolecule.getID());
+            throw new RuntimeException(anException);
+        }
+        //</editor-fold>
+        //
+        //<editor-fold desc="CycleFinder (Single Rings)" defaultstate="collapsed">
+        CycleFinder tmpMCBCycleFinder = Cycles.mcb();
+        IRingSet tmpMCBCyclesSet;
+        try {
+            Cycles tmpMCBCycles = tmpMCBCycleFinder.find(aMolecule);
+            tmpMCBCyclesSet = tmpMCBCycles.toRingSet();
+            for (IAtomContainer tmpContainer: tmpMCBCyclesSet.atomContainers()) {
+                for (IAtom tmpSingleRingAtom: tmpContainer.atoms()) {
+                    int tmpAtomInteger = tmpSingleRingAtom.getProperty(tmpAlkylSFAtomIndexProperty);
+                    tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISINRING,true);
+                    tmpAtomArray[tmpAtomInteger].setFlag(CDKConstants.ISPLACED,false);
+                }
+                for (IBond tmpSingleRingBond: tmpContainer.bonds()) {
+                    int tmpBondInteger = tmpSingleRingBond.getProperty(tmpAlkylSFBondIndexProperty);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISINRING,true);
+                    tmpBondArray[tmpBondInteger].setFlag(CDKConstants.ISPLACED,false);
+                }
+            }
+        } catch (Intractable e) {
+            throw new RuntimeException(e);
+        }
+        //</editor-fold>
+        //
+        //<editor-fold desc="Fragment Extraction" defaultstate="collapsed">
+        try {
+            IAtomContainer tmpRingFragmentationContainer = new AtomContainer();
+            IAtomContainer tmpChainFragmentationContainer = new AtomContainer();
+            //VISITED flag is set by ConjPiSysDetector, needs to be reset
+            for (IAtom tmpVisAtom : tmpAtomArray) {
+                tmpVisAtom.setFlag(CDKConstants.VISITED, false);
+            }
+            for (IBond tmpVisBond : tmpBondArray) {
+                tmpVisBond.setFlag(CDKConstants.VISITED, false);
+            }
+            //atom extraction
+            //superior performance compared to normal for iteration over Array length
+            for (IAtom tmpAtom : tmpAtomArray) {
+                System.out.println(tmpAtom.getProperty(tmpAlkylSFAtomIndexProperty).toString() + ": " + Arrays.toString(tmpAtom.getFlags()));
+                if (!tmpAtom.getFlag(CDKConstants.VISITED)) {
```

Comment:
In lines 590-592 you have set this flag to false for all atoms, so this check is not necessary. Further, you are using a for-each-statement, so each atom of the array will only be passed through once anyway. So I don't understand the point behind this construct.
Same for the bond arrays